### PR TITLE
Added LWP Leave Type

### PR DESCRIPTION
**Pull request by tejaszambre**
- Added new type as LWP in leave types
- Made necessory changes for the same in user_mailer and its view related to leave application
- Added leave type column in leave application, employee on leave tomorrow mails
- Removed WFH leaves from previous and upcoming leaves in leave application mail
- Added scope for filtering leave application where leave type is Leave or LWP
- Updated the Testcases for the above

### CODE REVIEW

**Comment by tejaszambre** [comment]
[LWP leave application.](https://trello.com/c/lWRXbt6b/34-lwp-leave-application)

File: spec/controllers/leaves_controller_spec.rb [hunk 1]
```diff
@@ -529,6 +529,50 @@
...
+      params = {
+        id: @leave.id,
+        leave_application: @leave.attributes.merge(leave_status: 'Approved')
+      }
+      put :update, params
+
+      leave_count = @user.reload.employee_detail.available_leaves
+      expect(flash[:success]).to eq('Your request has been updated successfully.' +
+        ' Please wait for the approval.')
+      expect(@leave.reload.leave_status).to eq('Approved')
```
Comment (Ashish944) [inline spec/controllers/leaves_controller_spec.rb:571]: Use constant APPROVED

File: spec/controllers/leaves_controller_spec.rb [hunk 2]
```diff
@@ -529,6 +529,50 @@
...
+      params = {
+        user_id: @user.id,
+        leave_application: @leave.attributes
+      }
+      post :create, params
+
+      leave_count = @user.reload.employee_detail.available_leaves
+      expect(flash[:success]).to eq('Your request has been submitted successfully.' +
+        ' Please wait for the approval.')
+      expect(@user.leave_applications.last.leave_status).to eq('Pending')
```
Comment (Ashish944) [inline spec/controllers/leaves_controller_spec.rb:555]: Use constant PENDING

File: app/mailers/user_mailer.rb [hunk 3]
```diff
@@ -22,20 +22,23 @@ def leave_application(sender_email, receivers, leave_application_id)
     @notification_names = @user.employee_detail.present? ? @user.employee_detail.get_notification_names : []
     @older_leaves = LeaveApplication.get_users_past_leaves(@user.id)
     @next_planned_leaves = LeaveApplication.get_users_upcoming_leaves(@user.id).where(
-      :id.ne => leave_application_id
+      'id.ne': leave_application_id
+    )
+    @leave_type = get_leave_type(
+      LeaveApplication.where(id: leave_application_id).first.leave_type
```
Comment (Ashish944) [inline app/mailers/user_mailer.rb:28]: @tejaszambre you could have added this var below the var leave_application and used that variable to pass the leave type

Comment (tejaszambre) [reply app/mailers/user_mailer.rb:28]: Ohh yes, will do that

File: spec/controllers/leaves_controller_spec.rb (hunk 1 above)
Comment (tejaszambre) [reply spec/controllers/leaves_controller_spec.rb:571]: Okay

File: spec/controllers/leaves_controller_spec.rb (hunk 2 above)
Comment (tejaszambre) [reply spec/controllers/leaves_controller_spec.rb:555]: Okay

**Comment by Ashish944** [comment]
Please check the below methods for LWP  - 
1. get_leaves_for_sending_reminder
2. approved_leaves_count

Also check leave application mails and rake task for the same LWP leave type

File: app/models/leave_application.rb [hunk 4]
```diff
@@ -49,6 +50,7 @@ class LeaveApplication
   scope :pending, ->{where(leave_status: PENDING)}
   scope :processed, ->{where(:leave_status.ne => PENDING)}
   scope :unrejected, -> { where(:leave_status.ne => REJECTED )}
+  scope :leaves, -> { where('$or': [{leave_type: LEAVE}, {leave_type: OPTIONAL}, {leave_type: LWP}]) }
```
Comment (Ashish944) [inline app/models/leave_application.rb:53]: Use leave_type.in instead of $or condition

File: app/models/time_sheet.rb [hunk 5]
```diff
@@ -1069,10 +1073,9 @@ def self.fetch_email_and_associate_to_user(user_id)
   end
 
   def self.approved_leaves_count(user, from_date, to_date)
-    user.leave_applications.where(
+    user.leave_applications.leaves.where(
       start_at: {'$gte': from_date, '$lte': to_date},
-      leave_status: LEAVE_STATUS[1],
-      leave_type: LeaveApplication::LEAVE
+      leave_status: APPROVED
```
Comment (Ashish944) [inline app/models/time_sheet.rb:1078]: This condition should contain only leave (LEAVE and LWP)

File: app/views/user_mailer/leave_application.html.haml [hunk 6]
```diff
@@ -74,6 +74,9 @@
                                                       %br/
                                                     %p
                                                       #{@user.public_profile.name} has applied for #{@leave_type} from #{@leave_application.start_at.to_s} to #{@leave_application.end_at.to_s}.
+                                                      - if @leave_application.is_leave?
+                                                        %br/
+                                                        Leave Type: #{@leave_application.leave_type}.
```
Comment (Ashish944) [inline app/views/user_mailer/leave_application.html.haml:79]: Remove this condition.

File: app/models/leave_application.rb (hunk 4 above)
Comment (tejaszambre) [reply app/models/leave_application.rb:53]: okay will do that

File: app/models/time_sheet.rb (hunk 5 above)
Comment (tejaszambre) [reply app/models/time_sheet.rb:1078]: okay will update that

File: app/views/user_mailer/leave_application.html.haml (hunk 6 above)
Comment (tejaszambre) [reply app/views/user_mailer/leave_application.html.haml:79]: okay I will update that

**Review by Ashish944 — APPROVED** [review]

**Comment by Sameer-Tilak** [comment]
LGTM

**Pull request merged**